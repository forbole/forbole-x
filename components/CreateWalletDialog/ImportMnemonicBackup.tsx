import { Button, DialogActions, DialogContent, Typography, Box, TextField } from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import PasswordInput from '../PasswordInput'
import useStyles from './styles'

interface ImportMnemonicBackupProps {
  onConfirm(params: { password: string; backupPhrase: string }): void
  error: string
}

const ImportMnemonicBackup: React.FC<ImportMnemonicBackupProps> = ({ onConfirm, error }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const [password, setPassword] = React.useState('')
  const [backupPhrase, setBackupPhrase] = React.useState('')

  return (
    <form
      noValidate
      onSubmit={(e) => {
        e.preventDefault()
        onConfirm({ password, backupPhrase })
      }}
    >
      <DialogContent className={classes.dialogContent}>
        <Typography>{t('secret recovery phrase backup')}</Typography>
        <TextField
          placeholder={t('secret recovery phrase backup')}
          variant="filled"
          InputProps={{
            disableUnderline: true,
          }}
          multiline
          rows={8}
          fullWidth
          value={backupPhrase}
          onChange={(e) => setBackupPhrase(e.target.value)}
        />
        <Box my={3}>
          <Typography>{t('encryption password')}</Typography>
          <PasswordInput
            placeholder={t('encryption password')}
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
          <Typography variant="body2" color="error">
            {error}
          </Typography>
        </Box>
      </DialogContent>
      <DialogActions>
        <Box flex={1} display="flex" flexDirection="column" mb={3}>
          <Button
            className={classes.button}
            variant="contained"
            color="primary"
            disabled={!password || !backupPhrase}
            type="submit"
          >
            {t('next')}
          </Button>
        </Box>
      </DialogActions>
    </form>
  )
}

export default ImportMnemonicBackup
