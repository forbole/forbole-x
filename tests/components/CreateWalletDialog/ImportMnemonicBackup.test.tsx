import React from 'react'
import renderer from 'react-test-renderer'
import last from 'lodash/last'
import ImportMnemonicBackup from '../../../components/CreateWalletDialog/ImportMnemonicBackup'

const onConfirm = jest.fn()
jest.mock('../../../components/PasswordInput', () => 'input')

describe('component: CreateWalletDialog - ImportMnemonicBackup', () => {
  it('renders correctly', () => {
    const component = renderer.create(<ImportMnemonicBackup onConfirm={onConfirm} error="" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly with error', () => {
    const component = renderer.create(<ImportMnemonicBackup onConfirm={onConfirm} error="error" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls onConfirm when button is clicked', () => {
    const component = renderer.create(<ImportMnemonicBackup onConfirm={onConfirm} error="" />)
    renderer.act(() => {
      component.root.findByType('textarea').props.onChange({
        target: {
          value: 'back up phrase',
        },
      })
      component.root.findByType('input').props.onChange({
        target: {
          value: 'encryption password',
        },
      })
    })
    renderer.act(() => {
      last(component.root.findAllByType('button')).props.onClick()
    })

    expect(onConfirm).toBeCalledWith({
      password: 'encryption password',
      backupPhrase: 'back up phrase',
    })
  })
})

afterEach(() => {
  jest.clearAllMocks()
})
