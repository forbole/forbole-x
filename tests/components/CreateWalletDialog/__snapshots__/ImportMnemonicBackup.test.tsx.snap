// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: CreateWalletDialog - ImportMnemonicBackup renders correctly 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      common:mnemonic phrase backup
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiFilledInput-multiline"
        onClick={[Function]}
      >
        <textarea
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMultiline MuiFilledInput-inputMultiline"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="common:mnemonic phrase backup"
          required={false}
          rows={8}
          value=""
        />
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-9"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        common:encryption password
      </p>
      <input
        onChange={[Function]}
        placeholder="common:encryption password"
        value=""
      />
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorError"
      >
        
      </p>
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-1 MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
      </button>
    </div>
  </div>
</form>
`;

exports[`component: CreateWalletDialog - ImportMnemonicBackup renders correctly with error 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      common:mnemonic phrase backup
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiFilledInput-multiline"
        onClick={[Function]}
      >
        <textarea
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMultiline MuiFilledInput-inputMultiline"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="common:mnemonic phrase backup"
          required={false}
          rows={8}
          value=""
        />
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-11"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        common:encryption password
      </p>
      <input
        onChange={[Function]}
        placeholder="common:encryption password"
        value=""
      />
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorError"
      >
        error
      </p>
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-12"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-1 MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
      </button>
    </div>
  </div>
</form>
`;
